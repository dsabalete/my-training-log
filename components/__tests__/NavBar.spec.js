import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import NavBar from '@/components/NavBar.vue'

const factory = () => {
  return shallowMount(NavBar, {
    stubs: {
      NuxtLink: RouterLinkStub
    }
  })
}

describe('NavBar', () => {
  let wrapper

  beforeEach(() => {
    wrapper = factory()
  })

  it('should be a Vue Instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('nav should have 3 links', () => {
    expect(wrapper.findAll('nav a').length).toBe(3)
  })
})
