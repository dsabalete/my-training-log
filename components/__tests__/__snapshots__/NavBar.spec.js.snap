// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders properly 1`] = `
"<div class=\\"nav\\"><a class=\\"brand\\">My Training Log</a>
  <nav><a>Types</a> |
    <a>Create</a> |
    <a>1RM</a></nav>
</div>"
`;
